### PR TITLE
feat: Do not mangle `AbortSignal` to avoid breaking `node-fetch`

### PR DESCRIPTION
**Pull request by kdy1**
### What?

Add `AbortSignal` to the reserved list of the name mangler.

### Why?

We don't want `node-fetch`  to throw an exception.

x-ref: https://vercel.slack.com/archives/C04KC8A53T7/p1700123182211619?thread_ts=1700090771.304599&cid=C04KC8A53T7

### How?

This PR modifies the mangle option from the Rust side code.

Closes PACK-1977

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/mangle-reserved | Change |
| - | - | - | - |
| buildDuration | 10.8s | 10.8s | N/A |
| buildDurationCached | 6.3s | 6.7s | ⚠️ +462ms |
| nodeModulesSize | 199 MB | 199 MB | ✓ |
| nextStartRea..uration (ms) | 404ms | 401ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/mangle-reserved | Change |
| - | - | - | - |
| 199-HASH.js gzip | 28.8 kB | 28.8 kB | N/A |
| 3f784ff6-HASH.js gzip | 53.3 kB | 53.3 kB | N/A |
| 494.HASH.js gzip | 180 B | 181 B | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 241 B | 239 B | N/A |
| main-HASH.js gzip | 31.7 kB | 31.8 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | ✓ |
| Overall change | 46.9 kB | 46.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/mangle-reserved | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/mangle-reserved | Change |
| - | - | - | - |
| _app-HASH.js gzip | 194 B | 195 B | N/A |
| _error-HASH.js gzip | 182 B | 181 B | N/A |
| amp-HASH.js gzip | 504 B | 506 B | N/A |
| css-HASH.js gzip | 322 B | 323 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | ✓ |
| edge-ssr-HASH.js gzip | 253 B | 255 B | N/A |
| head-HASH.js gzip | 348 B | 347 B | N/A |
| hooks-HASH.js gzip | 369 B | 368 B | N/A |
| image-HASH.js gzip | 4.3 kB | 4.3 kB | N/A |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.63 kB | 2.63 kB | N/A |
| routerDirect..HASH.js gzip | 311 B | 311 B | ✓ |
| script-HASH.js gzip | 384 B | 383 B | N/A |
| withRouter-HASH.js gzip | 307 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.17 kB | 3.17 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/mangle-reserved | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 483 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/mangle-reserved | Change |
| - | - | - | - |
| index.html gzip | 528 B | 526 B | N/A |
| link.html gzip | 538 B | 539 B | N/A |
| withRouter.html gzip | 524 B | 521 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/mangle-reserved | Change |
| - | - | - | - |
| edge-ssr.js gzip | 92.6 kB | 92.6 kB | N/A |
| page.js gzip | 145 kB | 145 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/mangle-reserved | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 625 B | 626 B | N/A |
| middleware-r..fest.js gzip | 150 B | 151 B | N/A |
| middleware.js gzip | 24.8 kB | 24.8 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 1.92 kB | 1.92 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/mangle-reserved | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 167 kB | 167 kB | ✓ |
| app-page-exp..prod.js gzip | 93.3 kB | 93.3 kB | ✓ |
| app-page-tur..prod.js gzip | 94.1 kB | 94.1 kB | ✓ |
| app-page-tur..prod.js gzip | 88.7 kB | 88.7 kB | ✓ |
| app-page.run...dev.js gzip | 137 kB | 137 kB | ✓ |
| app-page.run..prod.js gzip | 88 kB | 88 kB | ✓ |
| app-route-ex...dev.js gzip | 23.8 kB | 23.8 kB | ✓ |
| app-route-ex..prod.js gzip | 16.4 kB | 16.4 kB | ✓ |
| app-route-tu..prod.js gzip | 16.4 kB | 16.4 kB | ✓ |
| app-route-tu..prod.js gzip | 16 kB | 16 kB | ✓ |
| app-route.ru...dev.js gzip | 23.2 kB | 23.2 kB | ✓ |
| app-route.ru..prod.js gzip | 16 kB | 16 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.64 kB | 9.64 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-turbo...prod.js gzip | 21.8 kB | 21.8 kB | ✓ |
| pages.runtim...dev.js gzip | 22.5 kB | 22.5 kB | ✓ |
| pages.runtim..prod.js gzip | 21.8 kB | 21.8 kB | ✓ |
| server.runti..prod.js gzip | 49 kB | 49 kB | ✓ |
| Overall change | 924 kB | 924 kB | ✓ |

</details>

</details>
Commit: ad12ca6809969bc5e7ba16c68f2c93f8e5058140

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 7e392ad98d41902e66f079b02d907192619cc9ad

`pnpm test-dev test/e2e/app-dir/app-compilation/index.test.ts` 
- app dir > HMR > should not cause error when removing loading.js

<details>
<summary>Expand output</summary>

  ● app dir › HMR › should not cause error when removing loading.js

    TIMED OUT: hello from new page

    hello from slow page

    undefined

      626 |
      627 |   if (hardError) {
    > 628 |     throw new Error('TIMED OUT: ' + regex + '\n\n' + content + '\n\n' + lastErr)
          |           ^
      629 |   }
      630 |   return false
      631 | }

      at check (lib/next-test-utils.ts:628:11)
      at Object.<anonymous> (e2e/app-dir/app-compilation/index.test.ts:44:11)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: packages/next-swc/crates/napi/src/minify.rs [hunk 1]
```diff
@@ -109,6 +112,10 @@ fn patch_opts(opts: &mut JsMinifyOptions) {
         inline: Some(TerserInlineOption::Num(2)),
         ..Default::default()
     });
+    opts.mangle = BoolOrDataConfig::from_obj(MangleOptions {
+        reserved: vec!["AbortSignal".into()],
```
Comment (sokra) [inline packages/next-swc/crates/napi/src/minify.rs:116]: Should we also put it in the reserved list for `terser` options?

Comment (kdy1) [reply packages/next-swc/crates/napi/src/minify.rs:116]: I think so, will do

**Comment by feedthejim** [comment]
Could we perhaps add a test to avoid a regression there?

**Comment by kdy1** [comment]
Do you know how I can add a test for server minification? Will adding an API route and using `node-fetch` from there work?

**Review by sokra — APPROVED** [review]

**Pull request merged**